roles/addons/nhc/tasks/main.yml

```diff
@@ -35,11 +35,14 @@
     group: root
     mode: 0644
```

Comment:
I think I would go even farther. Adding here a conditional to not generate nhc.conf if it is listed in the list above. So it allows advanced users to fully manipulate NHC.

```
when:
  - 'nhc.conf' not in (nhc_files | default([]) | selectattr('name','defined') | map(attribute='name') | list)
```

(Untested)

Reply:
This task implements a similar mechanism to template nhc.conf with `nhc_static_configuration`, in a simpler way, documented in *General usage*.
I don't consider *nhc.conf* eligible to use in the task `template █ Copy nhc custom files` (documented in *Advanced usage). If one decide to do this, they will have an idempotency issue. To prevent this, I would tend to ensure `nhc.conf` is not allowed as a name in `nhc_files`.

Reply:
I didn't explained well what I want.

Since you implemented the capability to push whole files, using file name and content, I would like to:

1. Keep previous nhc.conf.j2 template, but remove the following lines inside it:

```
{% if nhc_static_configuration is defined and nhc_static_configuration is not none %}
{{ nhc_static_configuration }}
{% else %}
```

So template files allows simple configuration for simple usage, aka default NHC parameters and `nhc_checks` variable to fill it.

2. Add a conditional on this nhc.conf.j2 template, with an:

```
when:
  - 'nhc.conf' not in (nhc_files | default([]) | selectattr('name','defined') | map(attribute='name') | list)
```

To disable this template generation if `nhc.conf` is present in the `nhc_files` list. (so remove idempotent related issues).


This way, users have the choice, and role is simpler: they can either use the basic usage, with `nhc_checks`, or use a fully customized one with `nhc_files` (or combine both if desired).

Reply:
Done